swarm/storage/chunker.go

```diff
@@ -160,43 +160,43 @@ func TreeSplit(data io.Reader, size int64, putter Putter) (k Address, wait func(
 }
 
 func NewTreeJoiner(params *JoinerParams) *TreeChunker {
-	self := &TreeChunker{}
-	self.hashSize = params.hashSize
-	self.branches = params.chunkSize / params.hashSize
-	self.addr = params.addr
-	self.getter = params.getter
-	self.depth = params.depth
-	self.chunkSize = params.chunkSize
-	self.workerCount = 0
-	self.jobC = make(chan *hashJob, 2*ChunkProcessors)
-	self.wg = &sync.WaitGroup{}
-	self.errC = make(chan error)
-	self.quitC = make(chan bool)
-
-	return self
+	tc := &TreeChunker{}
```

Comment:
I am wondering if it could be even nicer like this:

```
func NewTreeJoiner(params *JoinerParams) *TreeChunker {
	return &TreeChunker{
		hashSize:    params.hashSize,
		branches:    params.chunkSize / params.hashSize,
		addr:        params.addr,
		getter:      params.getter,
		depth:       params.depth,
		chunkSize:   params.chunkSize,
		workerCount: 0,
		jobC:        make(chan *hashJob, 2*ChunkProcessors),
		wg:          &sync.WaitGroup{},
		errC:        make(chan error),
		quitC:       make(chan bool),
	}
}
```

maybe even more performant...

Just a suggestion. Maybe out of this PR scope.